### PR TITLE
fix: plasma finalize

### PR DESCRIPTION
**Pull request by 0xyjk**
**Description**

When one block has multiple plasma commitment, the `ExpireChallenges` for loop will only execute once, which cause `activeComms` didn't get poped, and the `s.finalized` will never get updated

### CODE REVIEW

**Comment by tynes** [comment]
We really should have a falling test case that is fixed by this patch

File: op-plasma/damgr_test.go [hunk 1]
```diff
@@ -21,6 +21,49 @@ func RandomData(rng *rand.Rand, size int) []byte {
...
+		bn     uint64 = 1
+		l1F           = &mockL1Fetcher{}
+		err    error
+	)
+	l1F.ExpectFetchReceipts(common.HexToHash(""), nil, types.Receipts{}, nil)
+	l1F.ExpectL1BlockRefByNumber(bn, eth.L1BlockRef{Number: bn}, nil)
+
+	// simulate one block has multiple commitments
+	state := NewState(logger, &NoopMetrics{})
+	state.SetInputCommitment([]byte("0x1"), bn, 0)
```
Comment (trianglesphere) [inline op-plasma/damgr_test.go]: It would be nice to have a non-zero challenge window here to more closely follow the delayed finalization.

File: op-plasma/damgr.go [hunk 2]
```diff
@@ -240,7 +240,7 @@ func (d *DA) isExpired(bn uint64) bool {
 // trigger a derivation reset.
 func (d *DA) AdvanceL1Origin(ctx context.Context, l1 L1Fetcher, block eth.BlockID) error {
 	// do not repeat for the same origin
-	if block.Number <= d.origin.Number {
```
Comment (trianglesphere) [inline op-plasma/damgr.go:243]: Why was this changed? It seems like we should only advance the origin if the next block is strictly greater.

**Review by trianglesphere — COMMENTED** [review]
Nice addition with the test. I confirmed that removing the fixes causes the test to fail.

File: op-plasma/damgr.go (hunk 2 above)
Comment (0xyjk) [reply op-plasma/damgr.go:243]: the `AdvanceL1Origin` function is called in `PlasmaDataSource.Next` function, and inside `PlasmaDataSource.Next`, it will call `fetcher.GetInput` -> `state.GetOrTrackChallenge` -> `SetInputCommitment` then push commitment to `activeComms` 

for one block multiple commitments case, `AdvanceL1Origin` will be called multiple times, if use `<=` instead of `<`, the `activeComms` will never get updated

**Comment by trianglesphere** [comment]
@0xyjk op-node tests are failing. I'm able to replicate inside `op-node` folder with `go test ./...`

**Comment by 0xyjk** [comment]
> @0xyjk op-node tests are failing. I'm able to replicate inside `op-node` folder with `go test ./...`

it's fixed @trianglesphere

**Comment by axelKingsley** [comment]
Howdy @0xyjk, just wanted to call out that the `damgr` and `dastate` components were significantly refactored recently to separate Commitment and Challenge tracking. Given that, I am not sure this change is still applicable. 

Additionally, it seems like these changes still have failing test -- tests within the `devnet-plasma` target are timing out, which I had been able to reproduce through running locally and relaunching these checks on github.

Finally, I noticed you removed `Once().` from the mocks in both plasma, and one in `mock_eth_client`. I am not sure what impact that may have on other tests.

If you think the stuck finalization issue is still a problem with the refactored code, could you please rebase this and we can take another look at it?
